pkg/config/config.go

```diff
@@ -4,15 +4,44 @@ import (
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
+	"os"
+	"path/filepath"
 )
 
+type State struct {
+	Config   Config
+	Services map[string]Service
+}
+
 type Config struct {
 	Password string `json:"password,omitempty"`
 	Username string `json:"username,omitempty"`
 	Port     uint16 `json:"port"`
 }
 
-func NewConfig(configPath string) (*Config, error) {
+type Service struct {
+	Name        string
+	Description string
+	DisplayName string
+	Tags        []string
+	Plans       map[string]Plan
+}
+
+type Plan struct {
+	Name        string `json:"name"`
+	Image       string `json:"image"`
+	Description string `json:"description"`
+}
+
+type ServiceDefinition struct {
+	Name        string   `json:"name"`
+	Description string   `json:"description"`
+	DisplayName string   `json:"display_name"`
+	Tags        []string `json:"tags"`
+	Plans       []Plan   `json:"plans"`
+}
```

Comment:
Did we create these structs to simplify the service config? or is there another reason?

With this, there needs to be another stage of filling in `brokerapi.Service` if we want to continue using the pivotal library. I am sure there is a good reason for this that I am not aware of. But what challenge are we trying to address?

Reply:
Yes we do have to fill it out. I have almost finished a commit to work on top of this branch to fill in Service in catalog. We do have to do some work in Catalog, but since it shouldn't be called too often I think its okay.